### PR TITLE
chore(Typings): improving and simplifying typings to match latest... …

### PR DESCRIPTION
**Pull request by NathanWalker**
This matches the typings to the latest alpha.31 found here: https://github.com/borisyankov/DefinitelyTyped/tree/master/angular2

Brings `Http` usage to use `Headers` class as specified in latest.

I need help with Pipes and bringing them up to date with latest. If anyone can help there, please let me know. Current `ts` warnings I see with this update...

```
WARNING in ../app/pipes/CapitalizePipe.ts
(10,14): Class 'CapitalizePipe' incorrectly implements interface 'Pipe'.
  Property 'onDestroy' is missing in type 'CapitalizePipe'. (2420)

WARNING in ../app/pipes/CapitalizePipe.ts
(34,12): Type 'CapitalizePipe' is not assignable to type 'Pipe'. (2322)

WARNING in ../app/pipes/RxPipe.ts
(2,28): Module '"angular2/angular2"' has no exported member 'ObservablePipe'. (2305)

WARNING in ../app/pipes/RxPipe.ts
(13,22): 'super' can only be referenced in a derived class. (2335)

WARNING in ../app/pipes/RxPipe.ts
(18,21): Property '_updateLatestValue' does not exist on type 'RxPipe'. (2339)

WARNING in ../app/pipes/RxPipe.ts
(26,32): Type 'RxPipe' is not assignable to type 'Pipe'.
  Property 'onDestroy' is missing in type 'RxPipe'. (2322)

WARNING in ../app/pipes/pipes.ts
(5,15): Module '"angular2/angular2"' has no exported member 'PipeRegistry'. (2305)

WARNING in ../app/pipes/pipes.ts
(15,9): Property 'append' does not exist on type 'typeof Pipes'. (2339)
```

### CODE REVIEW

**Comment by NathanWalker** [comment]
I was able to resolve the warnings above, but still trying to get Pipes to work with the latest api changes.

**Comment by gdi2290** [comment]
@NathanWalker if all the warnings are gone then let's merge it

File: src/app/pipes/RxPipe.ts [hunk 1]
```diff
@@ -20,6 +20,12 @@ export class RxPipe extends ObservablePipe {
       e => { throw e; }
     );
   }
+  transform(value: any, args: List<any> = null): any {
+    return;
```
Comment (gdi2290) [inline src/app/pipes/RxPipe.ts]: could you 
`return super. transform(value, args);`

File: src/app/pipes/RxPipe.ts [hunk 2]
```diff
@@ -20,6 +20,12 @@ export class RxPipe extends ObservablePipe {
...
+  }
+  onDestroy(): void {
+    // not sure what we should do here
```
Comment (gdi2290) [inline src/app/pipes/RxPipe.ts]: same here

`return super.onDestroy();`

File: src/app/pipes/CapitalizePipe.ts [hunk 3]
```diff
@@ -13,15 +13,18 @@ export class CapitalizePipe implements Pipe {
   supports(txt): boolean {
     return isString(txt);
   }
-  transform(input: string, isEveryWord?: boolean): string {
-    return (!input) ? '' :
-      (!isEveryWord) ?
-        this.capitalizeWord(input) :
-        input.replace(this.regexp, this.capitalizeWord);
+  transform(value: string, args: List<any>): any {
```
Comment (gdi2290) [inline src/app/pipes/CapitalizePipe.ts]: the second argument is either optional or null

`args?: List<any>` or `args: List<any> = null`

File: src/app/pipes/CapitalizePipe.ts [hunk 4]
```diff
@@ -13,15 +13,18 @@ export class CapitalizePipe implements Pipe {
...
+    return (!value) ? '' :
+      (!args) ?
+        this.capitalizeWord(value) :
+        value.replace(this.regexp, this.capitalizeWord);
   }
   capitalizeWord(txt: string): string {
     return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
   }
+  onDestroy(): void {
+    // not sure what we should do here
```
Comment (gdi2290) [inline src/app/pipes/CapitalizePipe.ts]: this is fine since it's stateless

**Comment by NathanWalker** [comment]
Ok @gdi2290 I have this updated per your comments, thanks!
However, I will mention although no warnings appear and everything appears to be solid, I'm having an issue with webpack (I think?) on my local system. It appears the __build__ folder is never created or is not served up... (not sure what that issue is, if related to changes or maybe just something on my system... have you seen this before?):
screen shot 2015-07-22 at 10 43 31 am

**Comment by NathanWalker** [comment]
@gdi2290 Any thoughts on the above... I'd like to see these changes go in but just want to verify things but am unable to at moment. Seems if I just run `npm start`, and I view the app at `http://localhost:8080` the `__build__` folder does load in the browser (however console shows I need the express server running). But the minute, in a seperate tab window, I run `npm run express` and then reload the app, it fails to load in the `__build__` folder, it just goes away and I get `404`'s in the browser on all the requested javascript build files. What gives?

**Comment by gdi2290** [comment]
@NathanWalker are you free to screenshare?

**Comment by NathanWalker** [comment]
@gdi2290 Just got back in, thanks for reaching out. Maybe in morning?

**Comment by NathanWalker** [comment]
@gdi2290 the solution you offered here:
https://github.com/angular-class/angular2-webpack-starter/issues/37#issuecomment-124897281
Will probably work. I'll try tonight or tomorrow sometime.

**Comment by jmac11** [comment]
I hope lol yeah it should be ok I got a usb play keyboard somewhere around I'll hunt it down and see if that will trigger an action

**Comment by NathanWalker** [comment]
Ok @gdi2290 That indeed did fix it so I updated the `package.json` to include that `NODE_ENV=development` when running the sample backend api server, as well as updated the README with a line on how to run it.

**Comment by NathanWalker** [comment]
Everything runs fine now (with app-simple.js).... when running with app.js, there are some TypeScript warnings related to latest 2.0.0-alpha.32 typings with `bind(Class).toClass(Class)`... which `toClass` expects `Type`. You may be able to help solve that if you pull down this PR and take a look. That's the last thing I'm stuck on.
Generates warnings like this...
```

WARNING in ./components/rxjs-examples/autosuggest/Github.ts
(29,24): Argument of type 'typeof Github' is not assignable to parameter of type 'Type'.
  Type 'Github' is not assignable to type 'Type'.
    Property 'apply' is missing in type 'Github'. (2345)
```

**Comment by gdi2290** [comment]
thanks!

**Pull request merged**